### PR TITLE
8258986: getColor throws IOOBE when PixelReader reads the same pixel twice

### PR DESCRIPTION
**Pull request by kevinrushforth**
As indicated in the JBS bug, using a `PixelReader` to read a scaled image in HiDPI mode, for example an `@2x` image, to read more than one pixel will read data from the wrong location in the image, usually leading to an IOOBE.

The bug is in the `getPixelAccessor` method of Prism Image. The method correctly creates a new `Accessor` if one hasn't already been created, but then it unconditionally wraps the current `Accessor` in a `ScaledPixelAccessor` if the scale is > 1. So the second time this method is called, it created another `ScaledPixelAccessor` that wraps the first `ScaledPixelAccessor`, meaning that the indexes into the pixel array are multiplied by 4. This continues for each new call to this method.

The fix is to only wrap an `Accessor` in a `ScaledPixelAccessor` the first time, when it is created.

I added a test, which is run for both scaled and unscaled images, to ensure that we get the right value when reading a pixel, and that reading it a second time returns the same result. Without the fix, the new test will fail with IOOBE when the scale factor is two. Related to this, I initially commented out the wrapping in a  `ScaledPixelAccessor` entirely, just to see what would happen, and none of the existing tests caught it. The new test will catch this.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258986](https://bugs.openjdk.java.net/browse/JDK-8258986): getColor throws IOOBE when PixelReader reads the same pixel twice

### Reviewers
 * [Ambarish Rapte](https://openjdk.java.net/census#arapte) (@arapte - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/389/head:pull/389`
`$ git checkout pull/389`

### CODE REVIEW

File: modules/javafx.graphics/src/main/java/com/sun/prism/Image.java [hunk 1]
```diff
@@ -650,9 +650,9 @@ public Image promoteByteRgbToByteBgra() {
                                                   (IntBuffer) pixelBuffer);
                     break;
             }
-        }
-        if (pixelaccessor != null && pixelScale != 1.0f) {
-            pixelaccessor = new ScaledAccessor<>(pixelaccessor, pixelScale);
+            if (pixelScale != 1.0f) {
+                pixelaccessor = new ScaledAccessor<>(pixelaccessor, pixelScale);
```
Comment (tomsontom) [inline modules/javafx.graphics/src/main/java/com/sun/prism/Image.java:654]: is that really correct? I think you should not overwrite save the scaled pixelAccessor in the instance variable - if I read that correct with this change the following happens: if I call getPixelAccessor() 3 times I get a ScaledAccessor wrapping a ScaleAccessor wrapping a ScaledAccessor, wrapping one of ByteAccess, ByteRgbAccess, IntAccess

Comment (kevinrushforth) [reply modules/javafx.graphics/src/main/java/com/sun/prism/Image.java:654]: No, that’s what used to happen before this fix.

The fix moves the wrapping of the pixelAccessor with a ScaledPixelAxcessor inside the null check so it’s now only done once.

Comment (tomsontom) [reply modules/javafx.graphics/src/main/java/com/sun/prism/Image.java:654]: ah I missed that the lines moved up inside the first if-check

**Review by arapte — APPROVED** [review]
+1 Test and sanity check looks good.

**Pull request closed without merging**